File: DataStructureJava/PriorityQueue.java
public class PriorityQueue<E extends Comparable<E>> implements Queue<E> {
    private MaxHeap<E> maxHeap;

    public PriorityQueue() {
        maxHeap = new MaxHeap<>();
    }

    @Override
    public int getSize() {
        return maxHeap.getSize();
    }

    @Override
    public boolean isEmpty() {
        return maxHeap.isEmpty();
    }

    @Override
    public E getFront() {
        return maxHeap.findMax();
    }

    // 时间复杂度O(logN)
    @Override
    public void enqueue(E e) {
        maxHeap.add(e);
    }

    // 时间复杂度O(logN)
    @Override
    public E dequeue() {
        return maxHeap.extractMax();
    }
}

File: DataStructureTypeScript/PriorityQueue.ts
import Queue from './interface/Queue'
import MaxHeap from './MaxHeap'

export default class PriorityQueue<E> implements Queue<E> {
  private maxHeap: MaxHeap<E> = new MaxHeap<E>()

  public getSize(): number {
    return this.maxHeap.size()
  }

  public isEmpty(): boolean {
    return this.maxHeap.isEmpty()
  }

  public getFront(): E {
    return this.maxHeap.findMax()
  }

  // 时间复杂度O(logN)
  public enqueue(e: E): void {
    this.maxHeap.add(e);
  }

  // 时间复杂度O(logN)
  public dequeue(): E {
    return this.maxHeap.extractMax();
  }
}